src/hotspot/share/gc/shared/c2/barrierSetC2.hpp

```diff
@@ -217,6 +217,12 @@ class BarrierSetC2: public CHeapObj<mtGC> {
   // expanded later, then now is the time to do so.
   virtual bool expand_macro_nodes(PhaseMacroExpand* macro) const { return false; }
   virtual void verify_gc_barriers(bool post_parse) const {}
+
```

Comment:
origin patch use has_load_barrier_data to replace has_load_barrier, why  we do not port it?

Reply:
```
-  virtual bool has_load_barriers() const { return false; }
+  virtual bool has_load_barrier_nodes() const { return false; }
```
the original patch rename `has_load_barriers` into `has_load_barrier_nodes`
because ZGC does not has C2 "load barrier nodes", while Shenandoah has this type of nodes.

We don't change it, because we want to make as little change as possible. Also, renaming can make it hard to isolate the change:
```
+#if !INCLUDE_ZGC
   virtual bool has_load_barriers() const { return false; }
-#else
+  virtual bool has_load_barrier_nodes() const { return false; }
+#endif
```
which messes up the code.

In conclusion, it's better to keep the original code.